src/caffe/layers/prelu_layer.cu

```diff
@@ -102,7 +107,7 @@ void PReLULayer<Dtype>::Backward_gpu(const vector<Blob<Dtype>*>& top,
             backward_buff_.gpu_diff(), multiplier_.gpu_data(), 1.,
             slope_diff);
       }
-    }
+
     if (channel_shared_) {
       caffe_gpu_add_scalar(this->blobs_[0]->count(), Dtype(dsum), slope_diff);
```

Comment:
After removing the loop, let's put merge this `if (channel_shared_)` into the `if (channel_shared_)` above.